sgt-back-end/index.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-console */
+
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:lfz@localhost/studentGradeTable'
+});
+
+app.get('/api/grades', (req, res, next) => {
+
+  const sql = `
+    select *
+    from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res, next) => {
+  const data = req.body;
+
+  if (data.name === '' || data.course === '' || data.score === '' || data.score < 0 || data.score > 100) {
```

Comment:
It is unlikely that the properties would be an empty string, you just need to check if the properties or truthy or not, which will also account for an empty string. For example:

```JAVASCRIPT
if (!data.name || !data.course || !data.score || data.score < 0 || data.score > 100)
```

Also I would break this up into 2 `if` statements, first check if any data is missing, then in a 2nd `if` check that the score is within range. This keeps the logic a bit simpler and allows you to send back more concise error messages.